### ISSUE TITLE
Error creating tables with NO_ZERO_DATE

### ISSUE THREAD

**Issue by rrennick** [woocommerce/woocommerce-admin#4104]
**Describe the bug**
Similar to https://github.com/woocommerce/action-scheduler/issues/519 creating the WC Admin custom tables throws errors in MySQL 5.7 when it's configured with

```
sql_mode=STRICT_TRANS_TABLES,NO_ZERO_DATE
```

Error log:
```
[09-Apr-2020 16:37:22 UTC] WordPress database error Invalid default value for 'date_created' for query ALTER TABLE wp_wc_admin_notes CHANGE COLUMN `is_snoozable` is_snoozable boolean DEFAULT 0 NOT NULL made by require('wp-cron.php'), do_action_ref_array('generate_category_lookup_table'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\CategoryLookup->regenerate, Automattic\WooCommerce\Admin\Install::create_tables, dbDelta, SharDB->query
[09-Apr-2020 16:38:23 UTC] WordPress database error Invalid default value for 'date_created' for query ALTER TABLE wp_wc_order_stats CHANGE COLUMN `returning_customer` returning_customer boolean DEFAULT NULL made by require('wp-admin/admin-ajax.php'), require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Install::check_version, Automattic\WooCommerce\Admin\Install::install, Automattic\WooCommerce\Admin\Install::create_tables, dbDelta, SharDB->query
[09-Apr-2020 16:38:23 UTC] WordPress database error Invalid default value for 'date_created' for query ALTER TABLE wp_wc_order_stats CHANGE COLUMN `customer_id` customer_id BIGINT UNSIGNED NOT NULL made by require('wp-admin/admin-ajax.php'), require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Install::check_version, Automattic\WooCommerce\Admin\Install::install, Automattic\WooCommerce\Admin\Install::create_tables, dbDelta, SharDB->query
[09-Apr-2020 16:38:23 UTC] WordPress database error Invalid default value for 'date_created' for query ALTER TABLE wp_wc_order_product_lookup CHANGE COLUMN `order_item_id` order_item_id BIGINT UNSIGNED NOT NULL made by require('wp-admin/admin-ajax.php'), require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Install::check_version, Automattic\WooCommerce\Admin\Install::install, Automattic\WooCommerce\Admin\Install::create_tables, dbDelta, SharDB->query
....
```

It looks like every table is affected by this.

**To Reproduce**

1. Install MySQL 5.7
2. Add the `sql_mode` above to your configuration
3. Drop the WC Admin tables
4. Delete the `woocommerce_admin_version` setting

**Expected behavior**
Two potential ways this could be fixed is modify the columns to allow NULL values and remove the defaults or change the defaults to be `1000-01-01 00:00:00`.
